File: DSA-Sheets/Striver-SDE-Sheet/01-Arr-I/Set matrix zero/row-col-track.cpp
// take 2 markers 1 row marker and 1 col marker all initialised zero
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    void setZeroes(vector<vector<int>> &matrix)
    {
        int n = matrix.size();
        int m = matrix[0].size();
        vector<int> row(n, 0);
        vector<int> col(m, 0);

        // mark the zeros in rowMarker and colMarker
        for (int i = 0; i < n; i++)
            for (int j = 0; j < m; j++)
                if (matrix[i][j] == 0)
                {
                    row[i] = 1;
                    col[j] = 1;
                }

        // iterate the matrix again and if marked make zero
        for (int i = 0; i < n; i++)
            for (int j = 0; j < m; j++)
                if (row[i] == 1 or col[j] == 1)
                    matrix[i][j] = 0;
    }
};

File: DSA-Sheets/Striver-SDE-Sheet/01-Arr-I/Set matrix zero/row-col-track.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        m = len(matrix)
        n = len(matrix[0])

        # make 2 lists to keep track of making row/col zero
        row_marking = [False] * m
        col_marking = [False] * n
        
        # iterate matrix rowwise and check for zeros 
        for row in range(m):
            for col in range(n):
                if matrix[row][col] == 0:
                    # take marking as True for zeroes
                    row_marking[row] = True
                    col_marking[col] = True

        # Traverse matrix again and look for marking
        for row in range(m):
            for col in range(n):
                # if marking found => mark that point as 0
                if row_marking[row] or col_marking[col]:
                    matrix[row][col] = 0